File: trunk/scripts/raw-to-c.pl
#!/usr/bin/perl -w

if ($#ARGV != 2)
{
	print "usage: raw-to-c <infile> <outfile> <name>\n";
	print "usage: raw-to-c reads the data from <filename> and creates\n";
	print "a const char[] with <name> in <outfile>\n";
	exit(1);
}

open INPUT, $ARGV[0] or die;
binmode INPUT;
open OUTPUT, ">$ARGV[1]" or die;

print OUTPUT "static const unsigned char $ARGV[2]\[\] = {\n";

my $buffer;
my $c;

while (my $size = read INPUT, $buffer, 20)
{
	print OUTPUT "\t";

	for (my $i = 0; $i < $size; $i++)
	{
		$c = vec $buffer, $i, 8;
		printf OUTPUT "0x%02x, ", $c;
	}
	print OUTPUT "\n";
}

print OUTPUT "};\n"

File: trunk/scripts/raw-to-c.py
#!/usr/bin/env python

import sys

chunk = 12
args = sys.argv[1:]

usage = """usage: raw-to-c <infile> <outfile> <name>

       raw-to-c reads the data from <infile> and creates
       a const char[] with <name> in <outfile>"""

if len(args) < 3:
    print usage
    sys.exit(1)

inf = open(args[0], 'rb')
outf = open(args[1], 'wb')

outf.write('static const unsigned char %s[] = {\n' % args[2])

data = inf.read(chunk)
while data:
    outf.write('    ')
    for c in data:
        outf.write('0x%02x, ' % ord(c))
    outf.write('\n')

    data = inf.read(chunk)
    
outf.write('};\n')

inf.close()
outf.close()
